### PR TITLE
feat: add a filter for unauthenticated response

### PR DESCRIPTION
**Pull request by labuchTractr**

### CODE REVIEW

**Review by GlennTractr — APPROVED** [review]

File: libs/nestjs/authentication/src/authentication.module.ts [hunk 1]
```diff
@@ -9,6 +10,7 @@ import {
   OPTIONS_TYPE,
 } from './authentication.module-definition';
 import { LoginController } from './controllers';
+import { UnAuthorizedExceptionFilter } from './filters';
```
Comment (floross) [inline libs/nestjs/authentication/src/authentication.module.ts]: `UnAuthorizedExceptionFilter` should be `UnauthorizedException` without the capital `A`

File: libs/nestjs/authentication/src/filters/unauthorized.filter.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const response = ctx.getResponse<Response>();
+    const request = ctx.getRequest<Request>();
+    const status = exception.getStatus();
+
+    response.cookie(this.cookieOptionsService.cookieName, '', {});
+    response.status(status).json({
+      statusCode: status,
+      timestamp: new Date().toISOString(),
+      path: request.url,
+    });
```
Comment (floross) [inline libs/nestjs/authentication/src/filters/unauthorized.filter.ts]: We should mimic the Error from nestjs Exception from https://docs.nestjs.com/exception-filters#built-in-http-exceptions. I know the exception exemple add timestamp and path but we should keep the json from the original error:

```json
{
  "message": "Something bad happened",
  "error": "Some error description",
  "statusCode": 400,
}
```

**Review by floross — CHANGES_REQUESTED** [review]

File: libs/nestjs/authentication/src/filters/unauthorized.filter.ts (hunk 2 above)
Comment (floross) [reply libs/nestjs/authentication/src/filters/unauthorized.filter.ts]: Probably you want to pass by the error message and the error form the original error

Comment (labuchTractr) [reply libs/nestjs/authentication/src/filters/unauthorized.filter.ts]: I put the name and the message from the exception in the body there is still the message error in the response as well

**Review by floross — APPROVED** [review]

**Pull request merged**